### DISCUSSION TITLE
How to access the Editor from the VimEditor

### DISCUSSION THREAD

**Discussion by jphalip** [JetBrains/ideavim#1032] [Q&A]
Hi,

I'm writing an extension for IdeaVim. My extension receives a `IJVimEditor` object which implements the `VimEditor` interface. I'd like to access its linked `com.intellij.openapi.editor.Editor` object. The issue is that, while the `IJVimEditor` class has a `getEditor()` method, the class is annotated as internal and therefore I cannot use it directly. The `VimEditor` interface is public but it does not have a `getEditor()` method.

Is there a workaround that I could use to access the `Editor` object?

Thank you.

**Comment by AlexPl292** [comment]
Hi! Can you try using `.ij` extension? Like: `vimEditor.ij`.

**Comment by jphalip** [comment]
Hi @AlexPl292 -- There doesn't seem to be an `ij` attribute or method available in the `vimEditor` object. I'm using Java by the way, not Kotlin.

**Comment by jphalip** [comment]
Here's a workaround I've come up with:

```java
            Window mostRecentFocusedWindow =
                WindowManager.getInstance().getMostRecentFocusedWindow();
            if (!(mostRecentFocusedWindow instanceof IdeFrame activeFrame)) return;
            Editor editor = FileEditorManager.getInstance(activeFrame.getProject()).getSelectedTextEditor();
```

It doesn't feel ideal though. It would be great if the `VimEditor.getEditor()` method could be made public/accessible.

**Reply by citizenmatt** [reply]
Yeah, don't do this 😄 You can use `IjVimEditorKt.getIj(vimEditor)` to access the `vimEditor.ij` extension property from Java.
